tests/unit/regression/lars_unittest.cc

```diff
@@ -233,6 +233,48 @@ TEST(LeastAngleRegression, lars_N_LessThan_D)
 	SG_UNREF(labels);
 }
 
+TEST(LeastAngleRegression, lars_n_less_than_d_32_feature_test)
+{
+	SGMatrix<float64_t> data_64(5,3);
+	SGVector<float64_t> lab(3);
+	generate_data_n_less_d(data_64,lab);
+
+	//copy data_64 into a float32_t SGMatrix
+	SGMatrix<float32_t> data(5,3);
+	for(index_t c = 0; c < data_64.num_cols; ++c)
+		for(index_t r = 0; r < data_64.num_rows; ++r)
+			data(r, c) = (float32_t) data_64(r, c);
+
+
+	CDenseFeatures<float32_t>* features = new CDenseFeatures<float32_t>(data);
```

Comment:
so now we need tests
 * for all other float types
 *`and for all other types (checking that errors are thrown)

@vigsterkr can we have a unit test that checks the train method for multiple feature types? And then we can just add the the classes that support multiple to that test, and specify which types it should work on

```
test_features_train_type(lars_instance, {float32_t, float64_6, floatmax_}) // checks that lars works with all those types and throws an error otherwise
```